exec/java-exec/src/main/java/org/apache/drill/exec/planner/FileSystemPartitionDescriptor.java

```diff
@@ -179,7 +179,7 @@ protected void createPartitionSublists() {
 
     // build a list of DFSDirPartitionLocation.
     for (final List<String> dirs : dirToFileMap.keySet()) {
-      locations.add( new DFSDirPartitionLocation((String [])dirs.toArray(), dirToFileMap.get(dirs)));
+      locations.add( new DFSDirPartitionLocation(dirs.toArray(new String [0]), dirToFileMap.get(dirs)));
```

Comment:
use `dirs.size()` instead of 0.

Reply:
According to https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size `.toArray(new T[0])` faster

Reply:
Thanks for the link. There is a difference in behavior of `ArrayList` and `Array.ArrayList` that is used in this method. In any case, if you touch this method, you will need to fix it, as there is no reason to create `ArrayList`, so I'd suggest to file a separate JIRA and fix it in a separate PR.